llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -21309,6 +21310,25 @@ bool RISCVTargetLowering::splitValueIntoRegisterParts(
     return true;
   }
 
+  if (ValueVT.isRISCVVectorTuple() && PartVT.isRISCVVectorTuple()) {
+    unsigned ValNF = ValueVT.getRISCVVectorTupleNumFields();
```

Comment:
Is this only called when PartVT and ValueVT are the same type? That shouldn't require any node to be created.

Reply:
I see, let me have another patch to fix it, thanks!